doc/stdenv/stdenv.chapter.md

```diff
@@ -206,6 +206,15 @@ For example, software using Wayland usually needs the `wayland` library at runti
 But it also executes the `wayland-scanner` program as part of the build to generate code, so `wayland` should also be added to `nativeBuildInputs`.
 :::
 
+::: {.note}
+When using native compilation, `stdenv` is lenient towards incorrect uses of `buildInputs` and `nativeBuildInputs`.
+That means a dependency needed at runtime often works, even if it is only present in `nativeBuildInputs`.
+Vice-versa, dependencies containing binaries that need to be executed during the build will work even if they are only listed in `buildInputs`.
+
+Add `strictDeps = true` as a parameter to `mkDerivation` or any of its language specific wrappers to disable this behavior.
+The specialized `build*` functions for dlang, go, nim, ocaml, python, and rust enable this option by default.
```

Comment:
We just enabled `strictDeps` by default for Emacs lisp packages: https://github.com/NixOS/nixpkgs/pull/379256.